src/main/webapp/app/tasks/search-field/search-field.component.ts

```diff
@@ -0,0 +1,29 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, OnInit, Output, } from '@angular/core';
```

Comment:
I guess there is no need because as far as understand this feature is intended for the users who have a URL with a search query in so that search string will be captured in the first route snapshot. I can update it if needed be in other use cases.